src/components/Snackbar/index.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { createContext, ReactNode, useCallback, useContext, useEffect, useState } from 'react';
+
+import 'react-toastify/dist/ReactToastify.css';
+import { ToastIcon } from 'react-toastify/dist/types';
+
+import { styled } from '@mui/material';
+import { toast, ToastContainer, ToastOptions } from 'react-toastify';
+
+type NotificationType = 'success' | 'error' | 'info' | 'warning';
+
+interface ToastNotificationProps {
+  message: string;
+  type: NotificationType;
+  options?: ToastOptions;
+  autoClose?: number | false;
+  icon?: ToastIcon | undefined;
+}
+
+interface NotificationContextProps {
+  showNotification: (props: ToastNotificationProps) => void;
+}
+
+export const NotificationContext = createContext<NotificationContextProps>({
+  showNotification: () => {},
+});
+
+export const NotificationProvider = ({ children }: { children: ReactNode }) => {
+  const [notification, setNotification] = useState<ToastNotificationProps | null>(null);
+
+  const showNotification = useCallback((props: ToastNotificationProps) => {
+    setNotification(props);
+  }, []);
+
```

Comment:
can we move notify out of the component? none of the variables it uses are from NotificationProvider